pyaedt/misc/toolkit_manager.py

```diff
@@ -0,0 +1,283 @@
+    install_button.grid(row=5, column=0, padx=5, pady=5, sticky="nsew")
+    uninstall_button = tk.Button(frame, text="Uninstall", bg="red", fg="white", padx=20, pady=5)
+    uninstall_button.grid(row=5, column=1, padx=5, pady=5, sticky="nsew")
+
+    def update_page(event=None):
+        selected_toolkit = toolkits_combo.get()
+        if selected_toolkit == "Custom":
+            install_button.config(text="Install")
+            uninstall_button.config(state="normal")
+        else:
+            if is_toolkit_installed(selected_toolkit):
+                install_button.config(text="Update")
+                uninstall_button.config(state="normal")
+            else:
+                install_button.config(text="Install")
+                uninstall_button.config(state="disabled")
+
+        if installation_option_action.get() == "Pip" and selected_toolkit != "Custom":
+            toolkit_name.config(state="disabled")
+            input_file_label.config(text="Enter wheelhouse path:")
+            input_file.config(state="disabled")
+        elif (installation_option_action.get() == "Pip" and selected_toolkit == "Custom") or (
+            installation_option_action.get() == "Offline" and selected_toolkit == "Custom"
+        ):
+            toolkit_name.config(state="normal")
+            input_file_label.config(text="Enter script path:")
+            input_file.config(state="normal")
+            installation_option_action.set("Offline")
+        else:
+            toolkit_name.config(state="disabled")
+            input_file_label.config(text="Enter wheelhouse path:")
+            input_file.config(state="normal")
+
+    toolkits_combo.bind("<<ComboboxSelected>>", update_page)
+
+    update_page()
+
+    return install_button, uninstall_button, input_file, toolkits_combo, toolkit_name
+
+
+def is_toolkit_installed(toolkit_name):
+    if toolkit_name == "Custom":
+        return False
+    script_file = os.path.normpath(os.path.join(package_dir, available_toolkits[toolkit_name]["toolkit_script"]))
+    return True if os.path.isfile(script_file) else False
+
+
+def open_window(window, window_name, open_source_toolkits):
+    """Open window action"""
+    if not hasattr(window, "opened"):
+        window.opened = True
+        window.title(window_name)
+        install_button, uninstall_button, input_file, toolkits_combo, toolkit_name = create_toolkit_page(
+            window, open_source_toolkits
+        )
+        root.minsize(500, 250)
+        return install_button, uninstall_button, input_file, toolkits_combo, toolkit_name
+    else:
+        window.deiconify()
+
+
+def __get_command_function(
+    is_install, toolkit_level, input_file, toolkits_combo, toolkit_name, install_button, uninstall_button
+):
+    return lambda: button_is_clicked(
+        is_install, toolkit_level, input_file, toolkits_combo, toolkit_name, install_button, uninstall_button
+    )
+
+
+def toolkit_window(toolkit_level="Project"):
+    """Main window"""
```

Comment:
This description should start with a verb that does not end in a "s" or "es". It also must conclude with a period.